addons/main/functions/fnc_EHKilled.sqf

```diff
@@ -25,21 +25,43 @@ private _holder = createVehicle ["WeaponHolderSimulated", (getPos _unit), [], 0,
 //add pack
 _holder addBackpackCargoGlobal [_chestpack, 1];
 private _backpack = firstBackpack _holder;
-clearAllItemsFromBackpack _backpack;
+[QGVAR(clearAllCargo), [_backpack]] call CBA_fnc_globalEvent;
+
 
 //add items
+private _cfgMagazines = configFile >> "CfgMagazines";
+private _cfgVehicles = configFile >> "CfgVehicles";
 {
     if (typeName (_x select 0) == "Array") then {
         //weapon with attachments
-        for "_i" from 0 to (_x select 1) do {
-            _backpack addWeaponWithAttachmentsCargoGlobal (_x select 0);
+        for "_i" from 1 to (_x select 1) do {
+            _backpack addWeaponWithAttachmentsCargoGlobal _x;
         };
     } else {
+        private _cargoClass = (_x select 0);
         //mags
-        if (isClass (configFile>>"CfgMagazines">> (_x select 0))) then {
+        if (isClass (_cfgMagazines >> _cargoClass)) then {
             _backpack addMagazineAmmoCargo _x;
+            [{
+                params ["_backpack", "_mag", "_count", "_rounds"];
+                private _countInBackpack = {_x isEqualTo [_mag, _rounds]} count magazinesAmmoCargo _backpack;
+                if (_countInBackpack < _count) then {
+                    _backpack addMagazineAmmoCargo [_mag, 1, _rounds];
+                };
+                _countInBackpack == _count
+            }, {}, [_backpack] + _x, 1, {
+                diag_log "bocr_main_fnc_EHKilled timed out adding magazines";
+                diag_log _this;
```

Comment:
```suggestion
               WARNING("EHKilled timed out adding magazines");
                TRACE_1("Container: ",_this);
```
Utilize CBA Macros instead of `diag_log`

Reply:
Thank you! Haven't really explored the debugging macros yet.